### PR TITLE
add ini file loading + ignore file & folders

### PR DESCRIPTION
**Pull request by Notenlish**
mvp

todo: raise warning for folders/files containing spaces
todo: add https://pygame-web.github.io/#configuration

### CODE REVIEW

**Comment by Notenlish** [comment]
example ini file:
```ini
[DEPENDENCIES]
ignoreDirs=ignore_folder
ignoreFiles=ignore_file.txt
```
if the user uses `"` or `'` it will raise an error and quit

File: src/pygbag/app.py [hunk 1]
```diff
@@ -379,6 +380,36 @@ async def main_run(app_folder, mainscript, cdn=DEFAULT_CDN):
 """
     )
 
+    config_parse = configparser.ConfigParser()
+    # config_parse["DEPENDENCIES"] = {"ignoreDirs":[], "ignoreFiles":[]}
+    if os.path.exists("pygbag.ini"):
+        config_parse.read("pygbag.ini")
+    else:
+        print("WARNING: No pygbag.ini found! See: https://pygame-web.github.io/#configuration")
```
Comment (pmp-p) [inline src/pygbag/app.py:388]: here's the place https://pygame-web.github.io/wiki/pygbag-configuration/

File: src/pygbag/filtering.py [hunk 2]
```diff
@@ -27,13 +25,17 @@
 /ATTIC
 """.strip().split(
     "\n"
-)
+)  # Why not .splitlines()
```
Comment (pmp-p) [inline src/pygbag/filtering.py:29]: splitlines() seems to have better cross platform support and micropython supports it when MICROPY_PY_BUILTINS_STR_SPLITLINES is on , so go for it !

File: src/pygbag/filtering.py [hunk 3]
```diff
@@ -16,24 +16,24 @@
...
 /ignore
 /static
 /ATTIC
-""".strip().split(
-    "\n"
-)
+""".splitlines()
 
-SKIP_EXT = ["lnk", "pyc", "pyx", "pyd", "pyi", "exe", "bak", "log", "blend", "DS_Store"]
+SKIP_EXT = ["pyc", "pyx", "pyd", "pyi", "exe", "bak", "log", "blend", "DS_Store"]
```
Comment (pmp-p) [inline src/pygbag/filtering.py:28]: why remove "lnk" from exclusion ?

File: src/pygbag/app.py [hunk 4]
```diff
@@ -379,6 +380,36 @@ async def main_run(app_folder, mainscript, cdn=DEFAULT_CDN):
...
+                raise SystemExit
+        
+        if ignore_raw.startswith("["):
+            if ignore_raw:
+                ignore_raw = ignore_raw.strip("[]")  # remove brackets
+                return [folder for folder in ignore_raw.split(",")]
+            else:
+                return []
+        else:  # just a single item without brackets
+            return [ignore_raw]
```
Comment (pmp-p) [inline src/pygbag/app.py:409]: why not just 
list(value.strip() for value in ignore_raw.splitlines())

```ini
;pygbag.ini
[DEPENDENCIES]
ignoreFiles =
    not_this_file1
    not_this_file2
    not_this_file3
```

File: src/pygbag/filtering.py (hunk 3 above)
Comment (Notenlish) [reply src/pygbag/filtering.py:28]: I dont know why lnk got removed, I probably deleted it accidently and forgot about it

File: src/pygbag/app.py (hunk 4 above)
Comment (Notenlish) [reply src/pygbag/app.py:409]: wouldnt ```ini
[DEPENDENCIES]
ignoreFiles = [file1, file2, file3]
```
be better? one file/folder per line would result in really long ini files

Comment (pmp-p) [reply src/pygbag/app.py:409]: i don't like [,] much it is not very friendly with output from "find"

File: src/pygbag/app.py [hunk 5]
```diff
@@ -380,35 +384,19 @@ async def main_run(app_folder, mainscript, cdn=DEFAULT_CDN):
 """
     )
 
-    config_parse = configparser.ConfigParser()
-    # config_parse["DEPENDENCIES"] = {"ignoreDirs":[], "ignoreFiles":[]}
     if os.path.exists("pygbag.ini"):
```
Comment (pmp-p) [inline src/pygbag/app.py:387]: i think that should better moved right after args processed here https://github.com/pygame-web/pygbag/blob/9da289f8e2840d5435dc3eeec6d67c3cdb66f134/src/pygbag/app.py#L406
 so this way with a flag eg "--ini" if there is no pygbag.ini we can create one with command line memorized in main section. That way next call could be just `pygbag main.py`  and ini will serve as some basic project file

File: src/pygbag/filtering.py
Comment (pmp-p) [inline src/pygbag/filtering.py:1]: anything starting with a dot should be ignored or keep .venv .tox in there

**Review by pmp-p — APPROVED** [review]
let's merge and improve later, there's so much to do with project files

**Pull request merged**